### DISCUSSION TITLE
Should useBlocker be used for blocking navigation to hash url

### DISCUSSION THREAD

**Discussion by Jarzka** [TanStack/router#1197] [Q&A]
Hi

Is the `useBlocker` hook meant to block navigation when navigating to hash urls on the same page?

Let's say that I'm on the page `/user` which contains a form. When I edit this form, I turn blocker navigation on because the page has come dirty (i.e. `useBlocker` fires if the user tries to navigate to another page like `/hello`. 

There is also a link on the page which navigates to `/user#about`. If I click this link, `useBlocker` is also fired, even if the navigation only happens on the same page. Is this the expected behavior of `useBlocker`? Is there a way to skip the firing if navigation happens on the same page via hash url?

**Comment by seedy** [comment]
I think your use case is worth an improvement of `useBlocker`, if possible, because hash navigation is not actual navigation here.

**Comment by robflowk** [comment]
I have a similar issue with two use cases:

- navigation to same route with different search params -> i see situations where blocking makes sense, and others where it does not. 
- A navigation to a child route with nested content -> my component will still be rendered, so i think this should not trigger a navigation block. If we navigate to a route that is not nested (a route file suffixed with _) however, this would not make sense.

maybe something like:
useBlocker(isDirty, {blockChildNavigation: true, blockSearchParams: true, blockHash: true})

 @tannerlinsley  What is your opinion?

**Comment by aliounebfam** [comment]
I have a similar issue with search params on the same route
Any update ?
